Apps/WebClient/src/NewClientApp/src/router/beforeEachGuard.ts

```diff
@@ -157,3 +160,43 @@ export const beforeEachGuard: NavigationGuard = async (
 
     next({ path: defaultPath });
 };
+
+function ticketIsValid(ticket: Ticket | undefined): boolean {
+    if (ticket?.status !== TicketStatus.Processed) {
+        return false;
+    }
+
+    const now = new Date().getTime();
+    return now < ticket.tokenExpires * 1000;
+}
+
+async function redirectWhenTicketIsInvalid(
+    to: RouteLocationNormalizedLoaded,
+    next: NavigationGuardNext
+) {
+    const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+    const waitlistStore = useWaitlistStore();
+
+    let ticket = waitlistStore.ticket;
+    try {
+        if (ticketIsValid(ticket)) {
+            logger.debug(`Router - check existing Processed ticket`);
+            await waitlistStore.scheduleCheckIn();
```

Comment:
```suggestion
            waitlistStore.scheduleCheckIn();
```
This doesn't need to be awaited because it doesn't return a Promise.